src/actions/dropbox/dropbox.ts

```diff
@@ -0,0 +1,211 @@
+        resp.success = false
+        resp.state = new Hub.ActionState()
+        resp.state.data = "reset"
+      })
+    } else {
+      resp.success = false
+      resp.message = "No data sent from Looker to be sent to Dropbox"
+    }
+    return resp
+  }
+
+  async form(request: Hub.ActionRequest) {
+    const form = new Hub.ActionForm()
+    form.fields = []
+
+    let accessToken = ""
+    if (request.params.state_json) {
+      try {
+        const jsonState = JSON.parse(request.params.state_json)
+        if (jsonState.code && jsonState.redirect) {
+          accessToken = await this.getAccessTokenFromCode(request)
+        }
+      } catch { winston.warn(request.params.state_json) }
+    }
+    const drop = this.dropboxClientFromRequest(request, accessToken)
+    return new Promise<Hub.ActionForm>((resolve, reject) => {
+      drop.filesListFolder({path: ""})
+        .then((resp) => {
+          form.fields = [{
+            description: "Dropbox directory where file will be saved",
+            label: "Save in",
+            name: "directory",
+            options: resp.entries.map((entries) => ({name: entries.name, label: entries.name})),
+            required: true,
+            type: "select",
+          }, {
+            label: "Filename",
+            name: "filename",
+            type: "string",
+          }]
+          if (accessToken !== "") {
+            const newState = JSON.stringify({access_token: accessToken})
+            form.state = new Hub.ActionState()
+            form.state.data = newState
+          } else {
+            form.state = new Hub.ActionState()
+          }
+          resolve(form)
+        })
+        .catch((_error: DropboxTypes.Error<DropboxTypes.files.ListFolderError>) => {
+          winston.info("Could not list Dropbox folders")
+          const actionCrypto = new Hub.ActionCrypto()
+          const jsonString = JSON.stringify({stateurl: request.params.state_url, app: request.params.appKey})
+          actionCrypto.encrypt(jsonString).then((ciphertextBlob: string) => {
+            form.state = new Hub.ActionState()
+            form.fields.push({
+              name: "login",
+              type: "oauth_link",
+              label: "Log in with Dropbox",
+              oauth_url: `${process.env.ACTION_HUB_BASE_URL}/actions/dropbox/oauth?state=${ciphertextBlob}`,
+            })
+            resolve(form)
+          }).catch((err: string) => {
+            winston.error("Encryption not correctly configured")
+            reject(err)
+          })
+        })
+    })
+  }
+  // async oauthUrl(redirectUri: string, stateUrl: string, encryptedState) {
+  async oauthUrl(redirectUri: string, encryptedState: string) {
+    const url = new URL("https://www.dropbox.com/oauth2/authorize")
+    const actionCrypto = new Hub.ActionCrypto()
+    const decryptedState = await actionCrypto.decrypt(encryptedState).catch((reason: any) => {
+      throw reason
```

Comment:
why the catch and rethrow? won't the `await` just throw?

Reply:
I'm guessing the linter requires the catch? That would be my guess but Wil is right that this is just catching and throwing immediately

Reply:
linter does require the catch. Adding more information here to make a better message